datastructures.md

```diff
@@ -0,0 +1,35 @@
+TrimDB Data Structures
+==
+
+TrimDB adopts a LSM[1] approach to data storage and retrieval.
+
+The TrimDB LSM approach comprises of several levels of data structure. Level c0 is special. It is an in-memory data structure that stores ordered key value pairs. Level c1 .. cn are Sorted String Tables (SSTables). SSTables are immutable on disk structures that contain keys and values, ordered by key.
```

Comment:
I think it should be in readme.md

Reply:
I tend to agree, @gra-moore you think you would like to move it directly into the readme?

Reply:
Yep sure.